### PR TITLE
NXP-28532: mention failing doc id and property when reindexing ES

### PR DESCRIPTION
**Pull request by NourNuxeo**
Helps with debugging when ES reindexing fails

### CODE REVIEW

File: modules/platform/nuxeo-elasticsearch/nuxeo-elasticsearch-core/src/main/java/org/nuxeo/elasticsearch/io/JsonESDocumentWriter.java [hunk 1]
```diff
@@ -212,7 +217,14 @@ protected static void writeProperties(JsonGenerator jg, DocumentModel doc, Strin
         }
 
         for (Property p : properties) {
-            writer.writeProperty(jg, p);
+            try {
+                writer.writeProperty(jg, p);
+            } catch (ClassCastException e) {
+                var nuxeoException = new NuxeoException(e);
+                nuxeoException.addInfo(
+                        String.format("writing JSON property failed on document: %s for property: %s", doc, p));
```
Comment (efge) [inline modules/platform/nuxeo-elasticsearch/nuxeo-elasticsearch-core/src/main/java/org/nuxeo/elasticsearch/io/JsonESDocumentWriter.java:225]: You can do directly `throw new NuxeoException(message, e)`, no need for `addInfo` which is only useful if you rethrow the same `NuxeoException` that you caught.

File: modules/platform/nuxeo-elasticsearch/nuxeo-elasticsearch-core/src/main/java/org/nuxeo/elasticsearch/io/JsonESDocumentWriter.java [hunk 2]
```diff
@@ -212,7 +217,12 @@ protected static void writeProperties(JsonGenerator jg, DocumentModel doc, Strin
         }
 
         for (Property p : properties) {
-            writer.writeProperty(jg, p);
+            try {
+                writer.writeProperty(jg, p);
+            } catch (ClassCastException e) {
+                throw new NuxeoException(
+                        String.format("writing JSON property failed on document: %s for property: %s", doc, p));
```
Comment (efge) [inline modules/platform/nuxeo-elasticsearch/nuxeo-elasticsearch-core/src/main/java/org/nuxeo/elasticsearch/io/JsonESDocumentWriter.java:224]: Please keep the original exception as last argument when throwing a new one.

Comment (NourNuxeo) [reply modules/platform/nuxeo-elasticsearch/nuxeo-elasticsearch-core/src/main/java/org/nuxeo/elasticsearch/io/JsonESDocumentWriter.java]: Yes sorry about that. Fixed

File: modules/platform/nuxeo-elasticsearch/nuxeo-elasticsearch-core/src/main/java/org/nuxeo/elasticsearch/io/JsonESDocumentWriter.java [hunk 3]
```diff
@@ -59,6 +62,8 @@
  */
 public class JsonESDocumentWriter {
 
+    private static final Logger log = LogManager.getLogger(JsonESDocumentWriter.class);
```
Comment (kevinleturc) [inline modules/platform/nuxeo-elasticsearch/nuxeo-elasticsearch-core/src/main/java/org/nuxeo/elasticsearch/io/JsonESDocumentWriter.java:65]: Doesn't seem used.

Comment (NourNuxeo) [reply modules/platform/nuxeo-elasticsearch/nuxeo-elasticsearch-core/src/main/java/org/nuxeo/elasticsearch/io/JsonESDocumentWriter.java:65]: yes not anymore. Removing 😬

**Pull request merged**